### PR TITLE
Added aliases

### PR DESCRIPTION
**Pull request by 8ctopus**
Thank you for a great library!

I implemented the aliases your way: https://github.com/thephpleague/iso3166/issues/61.

### CODE REVIEW

**Comment by 8ctopus** [comment]
Hello all, what's the status of this pull request? It's been hanging for 6 months.

**Comment by alcohol** [comment]
Can you fix the phpstan issues? Primarily the parameter types in the new aliases class.

File: src/ISO3166WithAliases.php [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+class ISO3166WithAliases implements ISO3166DataProvider
+{
+    private ISO3166DataProvider $source;
+
+    public function __construct(ISO3166DataProvider $iso3166)
+    {
+        $this->source = $iso3166;
+    }
+
+    public function name($name): array
```
Comment (alcohol) [inline src/ISO3166WithAliases.php]:
```suggestion
    public function name(string $name): array
```

File: src/ISO3166WithAliases.php [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+            if (0 === strcasecmp($alias, $name)) {
+                $name = $full;
+                break;
+            }
+        }
+
+        return $this->source->name($name);
+    }
+
+    public function alpha2($alpha2): array
```
Comment (alcohol) [inline src/ISO3166WithAliases.php]:
```suggestion
    public function alpha2(string $alpha2): array
```

File: src/ISO3166WithAliases.php [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    {
+        return $this->source->alpha2($alpha2);
+    }
+
+    public function alpha3($alpha3): array
```
Comment (alcohol) [inline src/ISO3166WithAliases.php]:
```suggestion
    public function alpha3(string $alpha3): array
```

File: src/ISO3166WithAliases.php [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    {
+        return $this->source->alpha3($alpha3);
+    }
+
+    public function numeric($numeric): array
```
Comment (alcohol) [inline src/ISO3166WithAliases.php]:
```suggestion
    public function numeric(string $numeric): array
```

File: tests/ISO3166WithAliasesTest.php [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    public $iso3166;
+
+    protected function setUp(): void
+    {
+        $this->iso3166 = new ISO3166WithAliases(new ISO3166);
+    }
+
+    public function testAlias(): void
+    {
+        $this->assertEquals($this->iso3166->name('United States')['name'], 'United States of America');
```
Comment (alcohol) [inline tests/ISO3166WithAliasesTest.php]:
```suggestion
        self::assertEquals($this->iso3166->name('United States')['name'], 'United States of America');
```

File: tests/ISO3166WithAliasesTest.php [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+        $this->assertEquals($this->iso3166->alpha2('US')['alpha2'], 'US');
```
Comment (alcohol) [inline tests/ISO3166WithAliasesTest.php]:
```suggestion
        self::assertEquals($this->iso3166->alpha2('US')['alpha2'], 'US');
```

File: tests/ISO3166WithAliasesTest.php [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+        $this->assertEquals($this->iso3166->alpha3('USA')['alpha3'], 'USA');
```
Comment (alcohol) [inline tests/ISO3166WithAliasesTest.php]:
```suggestion
        self::assertEquals($this->iso3166->alpha3('USA')['alpha3'], 'USA');
```

File: tests/ISO3166WithAliasesTest.php [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+        $this->assertEquals($this->iso3166->numeric('840')['numeric'], '840');
```
Comment (alcohol) [inline tests/ISO3166WithAliasesTest.php]:
```suggestion
        self::assertEquals($this->iso3166->numeric('840')['numeric'], '840');
```

File: tests/ISO3166WithAliasesTest.php [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+namespace League\ISO3166;
+
+use League\ISO3166\Exception\DomainException;
+use League\ISO3166\Exception\OutOfBoundsException;
+use PHPUnit\Framework\TestCase;
+
+class ISO3166WithAliasesTest extends TestCase
+{
+    /** @var ISO3166WithAliases */
+    public $iso3166;
```
Comment (alcohol) [inline tests/ISO3166WithAliasesTest.php:22]:
```suggestion
    public ISO3166DataProvider $iso3166;
```

File: tests/ISO3166WithAliasesTest.php [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+
+namespace League\ISO3166;
+
+use League\ISO3166\Exception\DomainException;
+use League\ISO3166\Exception\OutOfBoundsException;
+use PHPUnit\Framework\TestCase;
+
+class ISO3166WithAliasesTest extends TestCase
+{
+    /** @var ISO3166WithAliases */
```
Comment (alcohol) [inline tests/ISO3166WithAliasesTest.php]:
```suggestion
```

File: tests/ISO3166WithAliasesTest.php [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+/*
+ * (c) Rob Bast <rob.bast@gmail.com>
+ *
+ * For the full copyright and license information, please view
+ * the LICENSE file that was distributed with this source code.
+ */
+
+namespace League\ISO3166;
+
+use League\ISO3166\Exception\DomainException;
```
Comment (alcohol) [inline tests/ISO3166WithAliasesTest.php:15]:
```suggestion
use League\ISO3166\ISO3166DataProvider;
use League\ISO3166\Exception\DomainException;
```

File: src/ISO3166WithAliases.php [hunk 12]
```diff
@@ -0,0 +1,75 @@
...
+ */
+
+namespace League\ISO3166;
+
+use League\ISO3166\ISO3166;
+use League\ISO3166\ISO3166DataProvider;
+
+class ISO3166WithAliases implements ISO3166DataProvider
+{
+    private ISO3166DataProvider $source;
```
Comment (alcohol) [inline src/ISO3166WithAliases.php]:
```suggestion
    /** @var ISO3166DataProvider */
    private $source;
```

File: tests/ISO3166WithAliasesTest.php [hunk 13]
```diff
@@ -0,0 +1,35 @@
...
+namespace League\ISO3166;
+
+use League\ISO3166\ISO3166DataProvider;
+use League\ISO3166\Exception\DomainException;
+use League\ISO3166\Exception\OutOfBoundsException;
+use PHPUnit\Framework\TestCase;
+
+class ISO3166WithAliasesTest extends TestCase
+{
+    public ISO3166DataProvider $iso3166;
```
Comment (alcohol) [inline tests/ISO3166WithAliasesTest.php]:
```suggestion
    /** @var ISO3166DataProvider */
    public $iso3166;
```

**Comment by 8ctopus** [comment]
Thank you!

**Pull request merged**